### PR TITLE
[IMP] mail: adjust scroll in thread "ASC" mode on load more messages

### PR DESCRIPTION
**Pull request by alexkuhn**

### CODE REVIEW

File: addons/mail/static/src/new/thread/thread.js [hunk 1]
```diff
@@ -16,16 +16,51 @@ export class Thread extends Component {
...
+        this.messagesRef = useRef("messages");
+        this.pendingLoadMore = false;
         this.loadMoreState = useVisible("load-more", () => this.loadMore());
-        onMounted(() => this.loadMore());
-        onPatched(() => this.loadMore());
+        this.oldestMessageId = null;
+        useSnapshot("messages", {
+            onWillPatch: () => {
+                return {
+                    hasMoreMsgsAbove:
```
Comment (oomsveta) [inline addons/mail/static/src/new/thread/thread.js:26]: not fond of abbreviations

File: addons/mail/static/src/new/thread/thread.js [hunk 2]
```diff
@@ -16,16 +16,51 @@ export class Thread extends Component {
...
+        onMounted(() => {
+            this.oldestMessageId = this.messaging.getThreadOldestMsg(this.thread);
+            this.loadMore();
+        });
         onWillStart(() => this.requestMessages(this.props.id));
         onWillUpdateProps((nextProps) => this.requestMessages(nextProps.id));
     }
 
+    get thread() {
+        return this.messaging.threads[this.props.id];
```
Comment (oomsveta) [inline addons/mail/static/src/new/thread/thread.js:53]: aren't we supposed to identify threads from the couple of their id and model?

File: addons/mail/static/src/new/utils.js [hunk 3]
```diff
@@ -191,3 +199,41 @@ export function useVisible(refName, cb) {
     }
     return state;
 }
+
+export function useSnapshot(refName, { onWillPatch: p_onWillPatch, onPatched: p_onPatched }) {
```
Comment (oomsveta) [inline addons/mail/static/src/new/utils.js]: naming suggestion: `useScrollSnapshot`? as it is, I couldn't understand the purpose of this hook without reading the code.
You may also want to add a minimum of documentation, as it seems to be a function we intend to reuse.

File: addons/mail/static/src/new/utils.js [hunk 4]
```diff
@@ -191,3 +199,41 @@ export function useVisible(refName, cb) {
...
+    onWillPatch(() => {
+        const el = ref.el;
+        Object.assign(snapshot, {
+            scrollHeight: el.scrollHeight,
+            scrollTop: el.scrollTop,
+            clientHeight: el.clientHeight,
+        });
+        Object.assign(snapshot, {
+            ...p_onWillPatch(),
+        });
```
Comment (oomsveta) [inline addons/mail/static/src/new/utils.js]: why not putting it all in the same Object.assign?

File: addons/mail/static/src/new/thread/thread.js (hunk 2 above)
Comment (alexkuhn) [reply addons/mail/static/src/new/thread/thread.js:53]: It's not yet the case, `thread.id` is currently the id of channel or the "local id" of thread, i.e. `model,id`.
So here `this.props.id` is the local id of chatter thread.

File: addons/mail/static/src/new/thread/thread.js (hunk 1 above)
Comment (alexkuhn) [reply addons/mail/static/src/new/thread/thread.js:26]: Without abbreviation -- `hasMoreMessagesAbove ` --it's much harder to read.

File: addons/mail/static/src/new/utils.js (hunk 4 above)
Comment (alexkuhn) [reply addons/mail/static/src/new/utils.js]: done

File: addons/mail/static/src/new/utils.js (hunk 3 above)
Comment (alexkuhn) [reply addons/mail/static/src/new/utils.js]: done

**Pull request merged**